File: ICSharpCode.Decompiler.Tests/TestCases/VBPretty/VBExplicitInterfaceImplementation.cs
using System;

public interface IHasNameCountChanged
{
	string Name { get; }

	int Count { get; set; }

	event EventHandler Changed;
}
public class VBExplicitInterfaceImplementation : IHasNameCountChanged
{
	public string TheName => null;

	string IHasNameCountChanged.Name => this.TheName;

	public int TheCount { get; set; }

	int IHasNameCountChanged.Count {
		get {
			//ILSpy generated this explicit interface implementation from .override directive in TheCount
			return this.TheCount;
		}
		set {
			this.TheCount = value;
		}
	}

	public event EventHandler SomethingChanged;

	event EventHandler IHasNameCountChanged.Changed {
		add {
			//ILSpy generated this explicit interface implementation from .override directive in SomethingChanged
			this.SomethingChanged += value;
		}
		remove {
			this.SomethingChanged -= value;
		}
	}
}

File: ICSharpCode.Decompiler.Tests/TestCases/VBPretty/VBExplicitInterfaceImplementation.vb
Imports System

Public Interface IHasNameCountChanged
	ReadOnly Property Name As String
	Property Count As Integer
	Event Changed As EventHandler
End Interface

Public Class VBExplicitInterfaceImplementation
	Implements IHasNameCountChanged

	Public ReadOnly Property TheName As String Implements IHasNameCountChanged.Name
		Get
			Return Nothing
		End Get
	End Property

	Public Property TheCount As Integer Implements IHasNameCountChanged.Count

	Public Event SomethingChanged As EventHandler Implements IHasNameCountChanged.Changed
End Class
